### PR TITLE
Adding PyAMG package.

### PR DESCRIPTION
**Pull request by benc303**

### CODE REVIEW

File: var/spack/repos/builtin/packages/py-pyamg/package.py [hunk 1]
```diff
@@ -0,0 +1,25 @@
...
+
+class PyPyamg(PythonPackage):
+    """PyAMG is a library of Algebraic Multigrid (AMG) solvers with
+    a convenient Python interface."""
+
+    homepage = "https://github.com/pyamg/pyamg"
+    url      = "https://github.com/pyamg/pyamg/archive/v4.0.0.zip"
+
+    # A list of GitHub accounts to notify when the package is updated.
+    maintainers = []
```
Comment (tldahlgren) [inline var/spack/repos/builtin/packages/py-pyamg/package.py]: Are you sure you don't want to add your GitHub account to indicate you are a maintainer of the package?

**Review by tldahlgren — APPROVED** [review]
Confirmed the version exists in the specified repository.

I do recommend a maintainer be added.

File: var/spack/repos/builtin/packages/py-pyamg/package.py (hunk 1 above)
Comment (benc303) [reply var/spack/repos/builtin/packages/py-pyamg/package.py]: I’m not the maintainer. Those are @lukeolson and @jbschroder. Or do you mean maintainer of this Spack package file?

Comment (tldahlgren) [reply var/spack/repos/builtin/packages/py-pyamg/package.py]: I mean of the Spack package.

Comment (benc303) [reply var/spack/repos/builtin/packages/py-pyamg/package.py]: OK, done.

Comment (tldahlgren) [reply var/spack/repos/builtin/packages/py-pyamg/package.py]: Great!  Thanks.

**Comment by tldahlgren** [comment]
Thank you for your contribution!

**Pull request merged**